### PR TITLE
validate Header changes

### PR DESCRIPTION
**Pull request by schakrad**

### CODE REVIEW

**Review by leoluz — CHANGES_REQUESTED** [review]
Please add unit tests validating the new functionality.

**Comment by schakrad** [comment]
Test cases included in server_test.go

File: server/server_test.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+		expectedResult int
+	}{
+		{testName: "All Headers and path params are sent and are matched successfully",
+			header:         map[string][]string{"Argocd-Application-Name": []string{"argo:test"}},
+			params:         map[string]string{"application": "test"},
+			queryParams:    map[string]string{},
+			expectedResult: 200,
+		},
+	}
+	for _, validTest := range validTests {
```
Comment (leoluz) [inline server/server_test.go:185]:
```suggestion
	for _, validTest := range validTests {
		validTest := validTest
```

File: server/server_test.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+		},
+		{testName: "Application name mismatch. Value from the header is different from the path param application.",
+			header:         map[string][]string{"Argocd-Application-Name": []string{"argo:test1"}},
+			params:         map[string]string{"application": "test"},
+			queryParams:    map[string]string{},
+			expectedResult: 400,
+		},
+	}
+	//test invalid scenarios
+	for _, invalidTest := range invalidTests {
```
Comment (leoluz) [inline server/server_test.go:160]:
```suggestion
	for _, invalidTest := range invalidTests {
		invalidTest := invalidTest
```

File: server/server_test.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+		expectedResult int
+	}{
+		{testName: "All the headers and query params sent and are matched successfully",
+			header:         map[string][]string{"Argocd-Application-Name": []string{"argo:test"}, "Argocd-Project-Name": []string{"default"}},
+			params:         map[string]string{},
+			queryParams:    map[string]string{"application_name": "test", "project": "default"},
+			expectedResult: 200,
+		},
+	}
+	for _, validTest := range validTests {
```
Comment (leoluz) [inline server/server_test.go:103]:
```suggestion
	for _, validTest := range validTests {
		validTest := validTest
```

File: server/server_test.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+		},
+		{testName: "Project name mismatch. Value from the header is different from the query param project.",
+			header:         map[string][]string{"Argocd-Application-Name": []string{"argo:test"}, "Argocd-Project-Name": []string{"defaults"}},
+			params:         map[string]string{},
+			queryParams:    map[string]string{"application_name": "test", "project": "default"},
+			expectedResult: 400,
+		},
+	}
+	//test invalid scenarios
+	for _, invalidTest := range invalidTests {
```
Comment (leoluz) [inline server/server_test.go:78]:
```suggestion
	for _, invalidTest := range invalidTests {
		invalidTest := invalidTest
```

File: server/server_test.go (hunk 3 above)
Comment (schakrad) [reply server/server_test.go:103]: Made the change.

File: server/server_test.go (hunk 4 above)
Comment (schakrad) [reply server/server_test.go:78]: Made the change.

File: server/server_test.go (hunk 2 above)
Comment (schakrad) [reply server/server_test.go:160]: Made the change.

**Review by leoluz — APPROVED** [review]
LGTM

**Pull request merged**